contracts/libraries/Safe130To141Migration.sol

```diff
@@ -0,0 +1,124 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+/* solhint-disable one-contract-per-file */
+pragma solidity >=0.7.0 <0.9.0;
+
+import {SafeStorage} from "../libraries/SafeStorage.sol";
+
+interface ISafe {
+    function setFallbackHandler(address handler) external;
+}
+
+/**
+ * @title Migration Contract for Safe Upgrade
+ * @notice This contract facilitates the migration of a Safe contract from version 1.3.0 to 1.4.1.
+ *         The older versions should use built-in upgrade methods.
+ */
+contract Safe130To141Migration is SafeStorage {
+    // Address of this contract
+    address public immutable MIGRATION_SINGLETON;
+
+    // Address of Safe contract version 1.4.1 Singleton
+    address public constant SAFE_141_SINGLETON = address(0x41675C099F32341bf84BFc5382aF534df5C7461a);
+
+    // Address of Safe contract version 1.4.1 Singleton (L2)
+    address public constant SAFE_141_SINGLETON_L2 = address(0x29fcB43b46531BcA003ddC8FCB67FFE91900C762);
+
+    // Address of Safe contract version 1.4.1 Compatibility Fallback Handler
+    address public constant SAFE_141_FALLBACK_HANDLER = address(0xfd0732Dc9E303f09fCEf3a7388Ad10A83459Ec99);
+
+    /**
+     * @notice Constructor
+     * @dev Initializes the migrationSingleton with the contract's own address.
+     */
+    constructor() {
+        MIGRATION_SINGLETON = address(this);
+
+        require(isContract(SAFE_141_SINGLETON), "Safe 1.4.1 Singleton is not deployed");
+        require(isContract(SAFE_141_SINGLETON_L2), "Safe 1.4.1 Singleton (L2) is not deployed");
+        require(isContract(SAFE_141_FALLBACK_HANDLER), "Safe 1.4.1 Fallback Handler is not deployed");
+    }
+
+    /**
+     * @notice Event indicating a change of master copy address.
+     * @param singleton New master copy address
+     */
+    event ChangedMasterCopy(address singleton);
```

Comment:
Weird naming mix here (singleton and master copy)

Reply:
I think the ChangedMasterCopy event comes from previous contract versions (it was also used in the example contract, that's where I took it from) and might not require any backend changes. And the singleton comes from our latest naming conventions. I agree it looks weird, but I had a reason for that 😂 Shall I change the event name?

Reply:
I would change the event name (and also the natspec description). Maybe we should check with the API team, but I don't think we rely on this event :thinking:

Reply:
Not a big problem to change it, but it would make things easier to keep the same event `event ChangedMasterCopy(address masterCopy)` for L2 indexing. Also there should be an event for the Fallback Handler so we can detect the change. Current event is `event ChangedFallbackHandler(address handler)`

By the way, tracing indexing expects `function changeMasterCopy(address _masterCopy)` and `function setFallbackHandler(address handler)` . It will require a lot of refactoring to detect the new master copy and the fallback handler from `migrate` or `migrateWithFallbackHandler`, as we would have to hardcode the migration contract and the versions, and modify our processor to take into account what contract has been called via delegate call.